src/mon/OSDMonitor.cc

```diff
@@ -10848,6 +10848,18 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     err = parse_erasure_code_profile(profile, &profile_map, &ss);
     if (err)
       goto reply;
+    if (profile_map.find("crush-failure-domain") != profile_map.end()) {
+      string failure_domain = profile_map["crush-failure-domain"];
+
+      int type = osdmap.crush->get_type_id(failure_domain);
```

Comment:
this is an anti-pattern of using `std::map`, you do the look twice using the same key.

```suggestion
    if (auto found = profile_map.find("crush-failure-domain");
        found != profile_map.end()) {
      const auto& failure_domain = found->second;
      int failure_domain_type = osdmap.crush->get_type_id(failure_domain);

```